### PR TITLE
[Snyk] reach lib bump 0.15

### PR DESCRIPTION
**Pull request by alexmylonas**
### Background

We want to bump @reach library to v0.15

### Changes

- bump package.json
- Updated changed types for reach utils

### Testing

- Functionality test locally

### CODE REVIEW

File: src/components/Dropdown/DropdownLink.tsx [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
-export const DropdownLink = forwardRefWithAs<DropdownLinkProps & { disabled?: boolean }, 'a'>(
-  function DropdownLink({ children, disabled, ...rest }, ref) {
-    return (
-      <ReachMenuLink disabled={disabled} ref={ref} {...rest}>
-        <MenuItem disabled={disabled}>{children}</MenuItem>
-      </ReachMenuLink>
-    );
-  }
-);
+export const DropdownLink: React.FC<DropdownLinkProps & { disabled?: boolean }> = ({
```
Comment (3nvi) [inline src/components/Dropdown/DropdownLink.tsx:7]:
```suggestion
export const DropdownLink: Polymorphic.ForwardRefComponent<'div', DropdownLinkProps & { disabled?: boolean }> = ({
```

File: src/components/Popover/PopoverTrigger.tsx [hunk 2]
```diff
@@ -1,29 +1,32 @@
...
-  const ref = useForkedRef(triggerRef, forwardedRef);
+  const ref = useComposedRefs(triggerRef, forwardedRef);
 
   return (
     <Comp
       ref={ref}
-      onMouseDown={wrapEvent(onMouseDown, toggle)}
-      onKeyDown={wrapEvent(onKeyDown, ({ key }) => key === 'Enter' && toggle())}
+      onMouseDown={toggle}
+      onKeyDown={({ key }: { key: string }) => key === 'Enter' && toggle()}
```
Comment (3nvi) [inline src/components/Popover/PopoverTrigger.tsx:22]: we still need to wrap them. The new method is called `composeEventHandlers` on the same package

File: src/components/Popover/PopoverTrigger.tsx [hunk 3]
```diff
@@ -1,29 +1,32 @@
...
 export type PopoverTriggerProps = NativeAttributes<'button'>;
 
-const PopoverTrigger = forwardRefWithAs<PopoverTriggerProps, 'button'>(function PopoverTrigger(
-  { as: Comp = 'button', onMouseDown, onKeyDown, ...rest },
-  forwardedRef
-) {
+const PopoverTrigger = React.forwardRef<
+  HTMLElement,
+  PopoverTriggerProps & { as: React.ElementType }
+>(function PopoverTrigger({ as: Comp = 'button', ...rest }, forwardedRef) {
```
Comment (3nvi) [inline src/components/Popover/PopoverTrigger.tsx:11]: Let's apply thee same changes as the ones above

**Review by 3nvi — CHANGES_REQUESTED** [review]
hey @alexmylonas , we need to make some changes to make all of this work. Before we even merge it, we need to install it in Panther and make sure that it passes our `npm run tsc` check.

You can see how you intsall a pounce branch to a project [here](https://medium.com/@jonchurch/use-github-branch-as-dependency-in-package-json-5eb609c81f1a)

File: src/components/Popover/PopoverTrigger.tsx (hunk 2 above)
Comment (alexmylonas) [reply src/components/Popover/PopoverTrigger.tsx:22]: Good callout

File: src/components/Dropdown/DropdownButton.tsx [hunk 4]
```diff
@@ -1,8 +1,11 @@
...
 import { MenuButton as ReachMenuButton } from '@reach/menu-button';
-import { ComponentWithAs } from '@reach/utils';
 import { NativeAttributes } from '../Box';
 
 export type DropdownButtonProps = NativeAttributes<'button'>;
-export const DropdownButton = ReachMenuButton as ComponentWithAs<'button', DropdownButtonProps>;
+export const DropdownButton = ReachMenuButton as React.ForwardRefRenderFunction<
+  'button',
+  DropdownButtonProps
+>;
```
[the remark this answers is not in the document]
Comment (alexmylonas) [reply src/components/Dropdown/DropdownButton.tsx]: Done!

File: src/components/Dropdown/DropdownItem.tsx [hunk 5]
```diff
@@ -1,18 +1,20 @@
...
-export const DropdownItem = forwardRefWithAs<DropdownItemProps & { disabled?: boolean }, 'div'>(
-  function DropdownItem({ children, onSelect = () => {}, disabled, ...rest }, ref) {
-    return (
-      <ReachMenuItem disabled={disabled} onSelect={onSelect} ref={ref} {...rest}>
-        <MenuItem disabled={disabled}>{children}</MenuItem>
-      </ReachMenuItem>
-    );
-  }
-);
+export const DropdownItem: React.FC<DropdownItemProps & { disabled?: boolean }> = ({
```
[the remark this answers is not in the document]
Comment (alexmylonas) [reply src/components/Dropdown/DropdownItem.tsx]: Updated!

File: src/components/Dropdown/DropdownLink.tsx (hunk 1 above)
Comment (alexmylonas) [reply src/components/Dropdown/DropdownLink.tsx]: Updated!

**Comment by alexmylonas** [comment]
I have installed branch on my local env. Related components seems to work & look as before!

**Review by 3nvi — APPROVED** [review]
If panther has no TS issues, then it LGTM 🎉

**Pull request merged**